src/Npgsql/NpgsqlActivitySource.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Npgsql.Internal;
+using System;
+using System.Diagnostics;
+using System.Net;
+
+namespace Npgsql
+{
+    static class NpgsqlActivitySource
+    {
+#if NET5_0_OR_GREATER
+        static readonly ActivitySource Source = new("Npgsql");
+
+        public static bool IsEnabled => Source.HasListeners();
+
+        public static IDisposable? CommandStart(NpgsqlConnector connector, string sql)
+        {
+            var settings = connector.Settings;
+            var activity = Source.StartActivity(settings.Database!, ActivityKind.Client);
+            if (activity is null)
+                return null;
+
+            activity.SetTag("db.system", "postgresql");
+            activity.SetTag("db.connection_string", connector.UserFacingConnectionString);
+            activity.SetTag("db.user", settings.Username);
+            activity.SetTag("db.name", settings.Database);
+            activity.SetTag("db.statement", sql);
+            activity.SetTag("db.connection.id", connector.Id);
```

Comment:
Is `db.connection.id` a standard OTel tag (I couldn't find docs), or just an extra one you're using. MySQL has a similar concept (i.e., "server thread") and I could see adding this to MySqlConnector.

It doesn't seem that OTel uses many multipart dotted names; should we call this `db.connection_id` instead?

Reply:
Just an extra: thought it might be useful for finding the specific connection on the backend.

> It doesn't seem that OTel uses many multipart dotted names; should we call this db.connection_id instead?

Sure, why not.

Reply:
:+1:

We should consider proposing that as part of the OTel standard (I have some notes I'm gathering for when I have time to get involved...).